src/shadowfax.py

```diff
@@ -0,0 +1,405 @@
+                        if before_curie and kedge_key not in before_curie_edges:
+                            before_curie_edges.append(kedge_key)
+                        elif kedge_key not in after_curie_edges:
+                            after_curie_edges.append(kedge_key)
+            aux_edges_list.append(aux_edges)
+            before_curie_edges_list.append(before_curie_edges)
+            after_curie_edges_list.append(after_curie_edges)
+        aux_edges_keys = []
+        before_curie_edges_keys = []
+        after_curie_edges_keys = []
+        for aux_edges in aux_edges_list:
+            sha256 = hashlib.sha256()
+            for x in set(aux_edges):
+                sha256.update(bytes(x, encoding="utf-8"))
+            aux_edges_key = sha256.hexdigest()
+            if aux_edges_key not in aux_edges_keys:
+                aux_graphs[aux_edges_key] = {"edges": list(aux_edges), "attributes": []}
+                aux_edges_keys.append(aux_edges_key)
+        for before_curie_edges in before_curie_edges_list:
+            sha256 = hashlib.sha256()
+            for x in set(before_curie_edges):
+                sha256.update(bytes(x, encoding="utf-8"))
+            before_curie_edges_key = sha256.hexdigest()
+            if before_curie_edges_key not in before_curie_edges_keys:
+                aux_graphs[before_curie_edges_key] = {"edges": list(before_curie_edges), "attributes": []}
+                before_curie_edges_keys.append(before_curie_edges_key)
+        for after_curie_edges in after_curie_edges_list:
+            sha256 = hashlib.sha256()
+            for x in set(after_curie_edges):
+                sha256.update(bytes(x, encoding="utf-8"))
+            after_curie_edges_key = sha256.hexdigest()
+            if after_curie_edges_key not in after_curie_edges_keys:
+                aux_graphs[after_curie_edges_key] = {"edges": list(after_curie_edges), "attributes": []}
+                after_curie_edges_keys.append(after_curie_edges_key)
+        main_edge = uuid.uuid1().hex
+        before_edge = uuid.uuid1().hex
+        after_edge = uuid.uuid1().hex
+        knowledge_graph["edges"][main_edge] = {
+            "subject": source_node,
+            "object": target_node,
+            "predicate": "biolink:related_to",
+            "sources": [
+                {
+                    "resource_id": "infores:shadowfax",
+                    "resource_role": "primary_knowledge_source",
+                }
+            ],
+            "attributes": [
+                {
+                    "attribute_type_id": "biolink:support_graphs",
+                    "value": aux_edges_keys
+                }
+            ]
+        }
+        knowledge_graph["edges"][before_edge] = {
+            "subject": source_node,
+            "object": curie,
+            "predicate": "biolink:related_to",
+            "sources": [
+                {
+                    "resource_id": "infores:shadowfax",
```

Comment:
Has this been added to infores?

Reply:
This was a debugging thing that slipped through the cracks, just to make it easier to differentiate which edges were actually coming from shadowfax. This has been changed to `infores:aragorn`